include/nuttx/net/net.h

```diff
@@ -478,6 +478,26 @@ FAR struct iob_s *net_ioballoc(bool throttled, enum iob_user_e consumerid);
 
 int net_checksd(int fd, int oflags);
 
+/****************************************************************************
+ * Name: net_clear_sinzero()
+ *
+ * Description:
+ *   In IPv4, sin_zero field exists in sockaddr_in for padding and should
+ *   be set to zero.
+ *
+ * Input Parameters:
+ *   from    - Address of source (may be NULL)
+ *   fromlen - The length of the address structure
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_NET_IPv4
```

Comment:
It doesn't make sense to just clear IPv4 address, we should clear all unused field to zero for all address family, otherwise the similiar error will report again and again.

Reply:
@xiaoxiang781216 In my understanding, padding field only exists in sockaddr_in (i.e. IPv4 family). So that's why I added ifdef CONFIG_NET_IPv4.

Reply:
BSD socket is a general interface, many protocol build on top of it: IPv4, IPv6, Unix domain, Bluetooth etc. Each protocol define his own address repsentation, and then may add the padding field like IPv4. We have to find a general solution.